client-fs.go

```diff
@@ -261,6 +261,92 @@ func (f *fsClient) ShareUpload(startsWith bool, expires time.Duration, contentTy
 	})
 }
 
+// readFile reads and returns the data inside the file located
+// at the provided filepath.
+func readFile(fpath string) (io.ReadCloser, error) {
+	// Golang strips trailing / if you clean(..) or
+	// EvalSymlinks(..). Adding '.' prevents it from doing so.
+	if strings.HasSuffix(fpath, "/") {
+		fpath = fpath + "."
+	}
+	fpath, e := filepath.EvalSymlinks(fpath)
+	if e != nil {
+		return nil, e
+	}
+	fileData, err := os.Open(fpath)
+	if err != nil {
+		return nil, err
+	}
+	return fileData, nil
+}
+
+// createFile creates an empty file at the provided filepath
+// if one does not exist already.
+func createFile(fpath string) (io.WriteCloser, error) {
+	st, e := os.Stat(fpath)
+	if e == nil {
+		if !st.Mode().IsRegular() { // If destination exists but is not regular.
+			return nil, e
```

Comment:
There is a problem here - this will crash since 'e == nil' and 'writeCloser == nil'